modules/core/src/test/java/org/apache/ignite/internal/processors/cache/persistence/IgniteDataStorageMetricsSelfTest.java

```diff
@@ -423,6 +482,66 @@ public void testWalCompressedBytes() throws Exception {
         assertCorrectWalCompressedBytesMetrics(n1);
     }
 
+    /**
+     * Populates a cache w/32 KB of data
+     *
+     * @param igniteEx  Node.
+     */
+    private void populateCache(IgniteEx igniteEx) {
+        for (int i = 0; i < 10; i++)
+            igniteEx.cache("cache").put(ThreadLocalRandom.current().nextLong(), new byte[(int)(32 * U.KB)]);
+    }
+
+    /**
+     * Turns off WAL logging.
+     *
+     * @param  igniteEx  Node.
+     * @throws Exception If failed.
+     */
+    private void turnOffWal(IgniteEx igniteEx) throws Exception {
+        WALDisableContext walDisableCtx = igniteEx.context().cache().context().walState().walDisableContext();
+        assertNotNull(walDisableCtx);
+
+        setFieldValue(walDisableCtx, "disableWal", true);
+
+        assertTrue(walDisableCtx.check());
+        assertNull(walMgr(igniteEx).log(new DataRecord(emptyList())));
+    }
+
+    /**
+     * Check the state of wal archive, and whether the total size of
+     * wal (and possibly wal archive) match what is expected.
+     *
+     * @param igniteEx       Node.
+     * @param hasWalArchive  Whether wal archiving is enabled.
+     * @throws Exception     If failed.
+     */
+    private void checkWalArchiveAndTotalSize(IgniteEx igniteEx, boolean hasWalArchive) throws Exception {
+        FileWriteAheadLogManager walMgr = walMgr(igniteEx);
+
+        SegmentRouter router = walMgr.getSegmentRouter();
+
+        assertEquals(router.hasArchive(), hasWalArchive);
+
+        //wait to avoid race condition where new segments(and corresponding .tmp files) are created after totalSize has been calculated.
+        if (router.hasArchive()) {
+            assertTrue(waitForCondition(
+                       () -> walMgr.walWorkFiles().length == igniteEx.configuration().getDataStorageConfiguration().getWalSegments(),
```

Comment:
Better to put in a variable:
`int expWalWorkSegements = igniteEx.configuration().getDataStorageConfiguration().getWalSegments();`

Reply:
Do not forget about the line length of 120 characters

Reply:
In general, you can not add a new method **FileWriteAheadLogManager#walWorkFiles**, but reuse the **FileWriteAheadLogManager#loadFileDescriptors**, make a method in the test that we will remake the **IgniteCheckedException** into **IgniteException**.

Reply:
done